keras_cv/layers/preprocessing/equalization.py

```diff
@@ -53,9 +53,15 @@ def equalize_channel(self, image, channel_index):
         histogram = tf.histogram_fixed_width(image, [0, 255], nbins=self.bins)
 
         # For the purposes of computing the step, filter out the nonzeros.
-        nonzero = tf.where(tf.not_equal(histogram, 0))
-        nonzero_histogram = tf.reshape(tf.gather(histogram, nonzero), [-1])
-        step = (tf.reduce_sum(nonzero_histogram) - nonzero_histogram[-1]) // (
+        # Zeroes are replaced by 1410065408 while calculating min to keep shape
+        # constant across input sizes for compatibility with vectorized_map
+        histogram_without_zeroes = tf.where(
+            tf.equal(histogram, 0),
+            tf.multiply(tf.ones((), tf.int32), 1410065408),
+            histogram,
+        )
+
+        step = (tf.reduce_sum(histogram) - tf.reduce_min(histogram_without_zeroes)) // (
```

Comment:
I don't think this update is exactly the same as the original implementation. 

The original nonzero_histogram was a vector (which can have dynamic length based on the input value), we would like to get a fix length nonzero_histogram. The `nonzero_histogram[-1]` in original implementation is not exactly same as the `tf.reduce_min(histogram_without_zeroes)`

Reply:
That makes sense. I referred to a different source that used the minimum non-zero value. The results seem to be similar.

Do you suggest I somehow find the last nonzero value from the histogram instead? I believe it can be done using a loop or something similar to ensure that an intermediate tensor doesn't interfere with `vectorized_map`.

Reply:
@LukeWood  for why we choose to use the last nonzero element in histogram. I would like to make sure we are mathematically correct.

Reply:
Yeah, we probably need to give this layer another look and add unit tests over the behavior.  This is a tricky layer.

Reply:
I am not 100% confident that we have the right implementation.

Reply:
The implementation I was aware of involves division of the histogram by the total number of pixels, followed by cumulative sum and multiplication with the highest possible value in the range (here, 255). Since this method seems to be different compared to what several open-source implementations follow, I have been completely referencing sources online. 

If there is any single source I should follow that explains the desired variation, I will code the approach in a follow-up commit. I have noticed all of these approaches correctly do the job of expanding the intensity range, however.

Reply:
@LukeWood are we keeping this PR on hold until #210 is resolved?

Reply:
Yeah, maybe I will be taking a stab at this tomorrow.   I think theres a good chance I drastically rewrite the layer.  It looks like the logic might actually be incorrect.

Reply:
> The implementation I was aware of involves division of the histogram by the total number of pixels, followed by cumulative sum and multiplication with the highest possible value in the range (here, 255). 

Do you want me to implement this method? Might save you some time @LukeWood

Reply:
https://github.com/keras-team/keras-cv/pull/234 will handle it - thanks for offering

Reply:
Done - Can you rebase

Reply:
(Rebase and retitle vectorize Equalization now that this is what this PR does)

Reply:
This is done! @LukeWood